From e3c5e563f5d52267e482d543274e89e630525afc Mon Sep 17 00:00:00 2001
From: Max S Kash <asukms@ya.ru>
Date: Fri, 17 May 2024 11:54:44 +0500
Subject: [PATCH] openssl: Fixed a bug with loading the engine

The external engine does not load (for example: gost_engine), the startup script does not find the configuration.

Signed-off-by: Max S Kash <asukms@ya.ru>
---
 Makefile                                 | 2 +-
 files/openssl.init                       | 4 ++--
 patches/150-openssl.cnf-add-engines-conf.patch    | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 3bb60bc5ed7531..048ec94d99e823 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openssl
 PKG_VERSION:=3.0.13
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 PKG_BUILD_FLAGS:=no-mips16 gc-sections no-lto
 
 PKG_BUILD_PARALLEL:=1
diff --git a/files/openssl.init b/files/openssl.init
index 1c1e8745ffeb40..0ce2dd18287794 100755
--- a/files/openssl.init
+++ b/files/openssl.init
@@ -32,8 +32,8 @@ enable_module() {
 		echo "If the engine was not built-in, remove 'config builtin' from /etc/config/openssl."
 	elif [ "$force" = 1 ]; then
 		printf "[Forced] "
-	elif ! grep -q "\\[ *$1_sect *]" /etc/ssl/modules.cnf.d/*; then
-		echo "$1: Could not find section [$1] in config files."
+	elif ! grep -q "\\[ *$1_sect *]" /etc/ssl/*.cnf.d/*; then
+		echo "$1: Could not find section [$1_sect] in config files."
 		return 1
 	elif [ "$builtin" = 1 ]; then
 		printf "[Builtin] "
diff --git a/patches/150-openssl.cnf-add-engines-conf.patch b/patches/150-openssl.cnf-add-engines-conf.patch
index 9fe9cdf590cdbf..202dfb4111118d 100644
--- a/patches/150-openssl.cnf-add-engines-conf.patch
+++ b/patches/150-openssl.cnf-add-engines-conf.patch
@@ -33,7 +33,7 @@ Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>
 +
 +[engines_sect]
 +.include /var/etc/ssl/engines.cnf
-+
++.include /etc/ssl/engines.cnf.d
 +.include /etc/ssl/modules.cnf.d
 +
  
